server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java

```diff
@@ -725,19 +719,30 @@ private TSExecuteStatementResp internalExecuteQueryStatement(String statement,
       if (plan instanceof AuthorPlan) {
         plan.setLoginUserName(username);
       }
+
+      TSExecuteStatementResp resp = null;
+
+      if (plan instanceof QueryPlan) {
+        resp = getQueryColumnHeaders(plan, username);
+      }
       // create and cache dataset
       QueryDataSet newDataSet = createQueryDataSet(queryId, plan);
+
+      if (plan instanceof ShowPlan || plan instanceof AuthorPlan) {
+        resp = getListDataResp(newDataSet);
+      }
```

Comment:
Rename this method as `getListDataSetHeaders` to keep the consistence.

Reply:
fixed